src/main/scala/io/qbeast/spark/table/IndexedTable.scala

```diff
@@ -498,10 +496,32 @@ private[table] class IndexedTableImpl(
       fraction: Double,
       options: Map[String, String]): Unit = {
     assert(fraction > 0d && fraction <= 1d)
-    val indexFiles = snapshot.loadIndexFiles(revisionID)
-    import indexFiles.sparkSession.implicits._
-    val files = indexFiles.transform(filterSamplingFiles(fraction)).map(_.path).collect()
-    optimize(files, options)
+    log.info(s"Selecting files to Optimize for Revision $revisionID")
+    // Load the Index Files for the given revision
+    val revisionFilesDF = snapshot.loadIndexFiles(revisionID)
+    import revisionFilesDF.sparkSession.implicits._
+    // Filter the Index Files by the fraction
+    if (isStaging(revisionID)) { // If the revision is Staging, we should INDEX the staged data up to the fraction
+      val bytesToOptimize = revisionFilesDF.map(_.size).collect().sum * fraction
```

Comment:
Use spark sql to find the value rather than collecting a sequence of Longs: `revisionFilesDS.agg(sum("size")).as[Long].first()`

Reply:
Or you can compute the total bytes and then the `bytesToOptimize` using `f` and log both values after `revisionFiles` are collected.

Reply:
true 🤡